### PR TITLE
python310Packages.yaramod: init at 3.12.2

### PR DESCRIPTION
**Pull request by msm-code**
###### Description of changes

YARA is a language used for creating signatures, especially for malware.
Yaramod is a library created by avast, for parsing and working with yara rules (signatures).
The main project page is https://github.com/avast/yaramod

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 211504` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python311Packages.yaramod</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python310Packages.yaramod</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Depends on #211703

**Comment by msm-code** [comment]
Sorry for obvious mistakes - I've fixed the description and changelog locally but apparently didn't push my ammended commit (or forgot `--force`).

Don't know how I missed the build fail for 3.11 - thanks!

File: pkgs/development/python-modules/yaramod/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  pythonImportsCheck = [
+    "yaramod"
+  ];
+
+  meta = with lib; {
+    description = " Parsing of YARA rules into AST and building new rulesets in C++";
+    homepage = "https://github.com/avast/yaramod";
+    changelog = "https://github.com/avast/yaramod/blob/v${version}/CHANGELOG.md";
+    license = with licenses; [ mit ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/yaramod/default.nix:46]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/yaramod/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "yaramod"
+  ];
+
+  meta = with lib; {
+    description = " Parsing of YARA rules into AST and building new rulesets in C++";
```
Comment (dotlambda) [inline pkgs/development/python-modules/yaramod/default.nix:43]:
```suggestion
    description = "Parsing of YARA rules into AST and building new rulesets in C++";
```

File: pkgs/development/python-modules/yaramod/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "avast";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-1q+lzNtm8qko9ZAxJjkmPOQjHD5GxB6YyEz0tr+QWGk=";
+  };
+
```
Comment (dotlambda) [inline pkgs/development/python-modules/yaramod/default.nix:23]:
```suggestion

  postPatch = ''
    rm -r deps
  '';

```
We try not to use vendored dependencies.

File: pkgs/development/python-modules/yaramod/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  dontUseCmakeConfigure = true;
+
+  buildInputs = [
+    libxcrypt
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
```
Comment (dotlambda) [inline pkgs/development/python-modules/yaramod/default.nix:33]:
```suggestion

  setupPyBuildFlags = [
    "--with-unit-tests"
  ];

```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please squash.

**Comment by msm-code** [comment]
Thanks for valuable comments. Squashed and pushed the results.

In the current state of this PR, the build fails. I'm in the process of reworking it to build without the vendored dependencies.

**Comment by msm-code** [comment]
I've rewritten the derivation to remove the vendored deps, and re-downloaded them myself.

I've tried to use versions already in nixpkgs, but the code seems to depend on the specific vendored versions (or in one case even patches them). For this reason this modification feels fragile to me (what if the project updates the vendored sources). But if that's the preferred solution, I'm OK with it.

I've tested the code with nixpkgs-review and in a nix-shell. I've also squashed my commit. Asked @dotlambda for re-review.

**Comment by msm-code** [comment]
Hi @dotlambda, what do you think about the current version?

I'll add once again that my preference would be to use vendored dependencies (to reduce the maintanance overhead when they are updated in the project - they are pinned to specific version and the project doesn't work with the version we have in nixpkgs).

Nevertheless, I've removed vendored deps as suggested. Let me know if you see anything else that can be improved.

File: pkgs/development/python-modules/yaramod/default.nix [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    repo = "pybind11";
+    rev = "refs/tags/v2.9.2";
+    hash = "sha256-O3bkexUBa+gfiJEM6KSR8y/iVqHqlCFmz/9EghxdIpw=";
+  };
+  json = fetchFromGitHub {
+    owner = "nlohmann";
+    repo = "json";
+    rev = "refs/tags/v3.9.1";
+    hash = "sha256-THordDPdH2qwk6lFTgeFmkl7iDuA/7YH71PTUe6vJCs=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yaramod/default.nix:29]: Why are we not using nixpkgs versions of json and pybind11?

Comment (msm-code) [reply pkgs/development/python-modules/yaramod/default.nix:29]: Honestly, not sure about pybind11. I tested it earlier and it was broken with nixpkgs vesion of pybind11. Now I've changed it to the nixpkgs version, and it passes all my tests (and the included pytest tests). So probably I was confused/tested a wrong version.

For json I didn't know how to get the "single header include" for it, but found an example in another package from nixpkgs, so I replaced that.

**Comment by msm-code** [comment]
Result of `nixpkgs-review pr 211504` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.yaramod</li>
    <li>python311Packages.yaramod</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.yaramod

**Pull request merged**